src/main/java/com/cloudant/http/interceptors/CookieInterceptor.java

```diff
@@ -121,6 +121,15 @@ private String getCookie(URL url) {
 
             HttpConnection conn = Http.POST(sessionURL, "application/x-www-form-urlencoded");
             conn.setRequestBody(sessionRequestBody);
+
+            //when we request the session we need all interceptors except this one
+
+            conn.requestInterceptors.addAll(context.connection.requestInterceptors);
+            conn.requestInterceptors.remove(this);
+            conn.responseInterceptors.addAll(context.connection.responseInterceptors);
+            conn.responseInterceptors.remove(this);
```

Comment:
I didn't have anything in mind, just wondered if its something we should do, or just let a user shoot themselves in the foot